### PR TITLE
feat: add jest coverage

### PR DESCRIPTION
**Pull request by apurv-wednesday**
## Summary by CodeRabbit

- **Chores**
  - Updated test configuration to enforce coverage thresholds and enhanced coverage reporting.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -53,4 +53,11 @@ jobs:
...
           lhci autorun
         env:
-          LHCI_GITHUB_APP_TOKEN: ${{ secrets.LHCI_GITHUB_APP_TOKEN }}
\ No newline at end of file
+          LHCI_GITHUB_APP_TOKEN: ${{ secrets.LHCI_GITHUB_APP_TOKEN }}
+
+      - name: Test and generate coverage report
+        uses: artiomtr/jest-coverage-report-action@v2.0-rc.4
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (alichherawalla) [inline .github/workflows/ci.yml:61]: use the latest version "v2.3.0"

Comment (alichherawalla) [reply .github/workflows/ci.yml:61]: "skip-step: all"

File: jest.config.ts [hunk 2]
```diff
@@ -6,6 +6,16 @@ const createJestConfig = nextJest({ dir: "./" });
 const jestConfig: Config = {
   preset: "ts-jest",
   testEnvironment: "jsdom",
+  // uncomment later after adding tests
```
Comment (alichherawalla) [inline jest.config.ts:9]: let's mimic the current coverage that the repo has. That way nothing that pushes the coverage below it can be committed.

**Review by alichherawalla — APPROVED** [review]

**Comment by alichherawalla** [comment]
one small change, merge post that

File: jest.config.ts (hunk 2 above)
Comment (apurv-wednesday) [reply jest.config.ts:9]: Cool. Understood

**Pull request merged**